python/cugraph/cugraph/structure/graph_implementation/simpleDistributedGraph.py

```diff
@@ -129,6 +170,37 @@ def __from_edgelist(
         self.source_columns = source
         self.destination_columns = destination
 
+        self.compute_renumber_edge_list(
+            transposed=False, legacy_renum_only=True
```

Comment:
I second what Rick mentioned. Furthermore, having renumbering at the graph creation(`__from_edgelist`)  forces to add `legacy_renum_only ` to `__from_edgelist` because some algos (the legacy algo) require the C++ renumbering and others(pylibcugraph, DGL) don't.

Reply:
Maybe I am missing something but I don't get why the edge list needs to be renumbered as `pylibcugraph` renumbers the edgelist internally when creating the `pylibcugraph` graph. Can you explain why?

Reply:
I did that because the original version of simpleDistributedGraph said renumbering was always required.  But I'll drop it and see what happens.

Reply:
I believe you are correct.  This was done just for consistency as some users may expect that the renumbering occurs.

Reply:
> I did that because the original version of simpleDistributedGraph said renumbering was always required'

Right,  renumbering([C++ renumbering](https://github.com/rapidsai/cugraph/blob/branch-22.08/python/cugraph/cugraph/structure/number_map.py#L590)) is `still` required for the algos following the legacy path like `MG louvain`(once we move everything to `pylibcugraph`, we will drop the C++ renumbering). In fact, those algos require the [aggregate_segment_offset](https://github.com/rapidsai/cugraph/blob/branch-22.08/python/cugraph/cugraph/structure/number_map.py#L650) which can only. be provided after computing the C++ renumbering

Reply:
Yes, so what I'm doing is keeping renumber=True as the default and adding an option for legacy renumbering, set to False by default.  I assume any user modifying those options knows what they're doing and is aware of the API differences.

Reply:
I added the `legacy_renum_only` to skip the `C++ renumbering` which is not needed for the `pylibcugraph` algos. It only does a `python` renumbering of the edgelist if the vertices are non integers. Sorry if the naming is confusing and should probably find a better name for that

Reply:
> I did that because the original version of simpleDistributedGraph said renumbering was always required. But I'll drop it and see what happens

I think if you drop it, it should still be find. But probably let's have a discussion regarding it during sync with the rest of the team

Reply:
Legacy_renum_only=False appears to break the pylibcugraph algorithms...

Reply:
Yes it needs to be set  to `True` for the `pylibcugraph` algos

Reply:
Well this causes a problem, basically the graph is now only usable on one API or the other depending on whether or not it's set to true.

Reply:
Credit to you for catching this; since all of our tests call only one of the APIs they did not fail.

Reply:
> Well this causes a problem, basically the graph is now only usable on one API or the other depending on whether or not it's set to true

Got it. We can have a post stand up on it after sync.

Reply:
I may have an idea of what is going on. Prior to this change, skipping the C++ algos was solely at the discretion of each algo which is no longer the case now. 

1. If you have `renumbering` in `__from_edgelist`, it will be skipped at the algo's call
2. We might also have to refactor the tests involved too.
I think the a solution could be do remove the renumbering in `__from_edgelist` but I am yet to test that